che-admin-api/che-core-api-plugin/src/main/java/org/eclipse/che/plugin/PluginsService.java

```diff
@@ -0,0 +1,233 @@
+
+    }
+
+    /**
+     * Deletes a plugin by providing its name
+     * @param pluginName the mandatory name of the plugin
+     * @return a descriptor if plugin has been found and deleted
+     * @throws PluginManagerException if there is a failure during removal
+     */
+    @DELETE
+    @Path("/{plugin-name}")
+    public PluginDescriptor deletePlugin(@PathParam("plugin-name") String pluginName) throws PluginManagerException {
+        return dtoBuilder.convert(pluginManager.removePlugin(pluginName));
+
+    }
+
+    /**
+     * Update plugin status. For example allowing to stage a plugin from available state to staged state
+     * @param pluginName the name of the plugin
+     * @param pluginAction the wanted action on the plugin
+     * @return description of the plugin with its new status
+     * @throws PluginManagerException if not able to get details of the plugin
+     */
+    @PUT
+    @Path("/{plugin-name}")
+    public PluginDescriptor updatePlugin(@PathParam("plugin-name") String pluginName, @QueryParam("action") PluginAction pluginAction)
+            throws PluginManagerException, PluginRepositoryException {
+        if (pluginName == null) {
+            throw new IllegalArgumentException("Should give non null plugin");
+        }
+        if (pluginAction == null) {
+            throw new IllegalArgumentException("Should give non null action");
+        }
+
+
+        final IPluginAction action;
+        try {
+            action = IPluginAction.valueOf(pluginAction.name());
+        } catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException(String.format("Invalid action given %s", pluginAction), e);
+        }
+
+
+        return dtoBuilder.convert(pluginManager.updatePlugin(pluginName, action));
+    }
+
+    /**
+     * Register/adds into Eclipse che a new plugin. It will be registered with available status
+     * @param plugin the reference of the plugin
+     * @return description of the plugin
+     * @throws PluginManagerException if not able to add the plugin
+     */
+    @POST
+    public PluginDescriptor addPlugin(@QueryParam("plugin") String plugin) throws PluginManagerException,
+                                                                                  PluginResolverNotFoundException {
+        if (plugin == null) {
+            throw new IllegalArgumentException("Should give non null plugin");
+        }
+        return dtoBuilder.convert(pluginManager.addPlugin(plugin));
+    }
+
+
+    /**
+     * Requires a new install of all staged plugins
+     * @return status of the current installation
+     */
+    @POST
+    @Path("/install")
+    public PluginInstall newInstall() throws PluginManagerException, PluginInstallerException {
+        return dtoBuilder.convert(pluginManager.requireNewInstall());
+    }
+
+    /**
+     * Gets the list of all install
+     * @return the IDs of every install
+     */
+    @GET
+    @Path("/install")
```

Comment:
it's to list all installations that have been done.
Like listing all processes, etc
It's not a verb

because we have /install and /install/<id>
like we could have /dogs/ and /dogs/<id>